src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -65,10 +66,27 @@ public ScanResults scan(ScanRequest scanRequest) {
         checkScanSubmitEmailDelivery(scanRequest);
 
         try {
-            Integer projectId;
+            Integer scanId;
             CxScanParams cxScanParams = scanRequestConverter.toScanParams(scanRequest);
-            Integer scanId = cxService.createScan(cxScanParams, "CxFlow Automated Scan");
-            projectId = cxScanParams.getProjectId();
+            Integer projectId = cxScanParams.getProjectId();
+
+            log.info("Checking if there is any existing scan for Project: {}", projectId);
+            Integer existingScanId = cxService.getScanIdOfExistingScanIfExists(projectId);
+
+            if (existingScanId != UNKNOWN_INT) {
+                if (flowProperties.getScanResubmit()) {
+                    log.info("Existing ongoing scan with id {} found for Project : {}", existingScanId, projectId);
+                    log.info("Aborting the ongoing scan with id {} for Project: {}", existingScanId, projectId);
+                    cxService.cancelScan(existingScanId);
+                    log.info("Resubmitting the scan for Project: {}", projectId);
+                    scanId = cxService.createScan(cxScanParams, CXFLOW_SCAN_MSG);
+                } else {
+                    log.warn("Property scan-resubmit set to {} : New scan not submitted, due to existing ongoing scan for the same Project id {}", flowProperties.getScanResubmit(), projectId);
+                    throw new CheckmarxException(String.format("Active Scan with Id %d already exists for Project: %d , ", existingScanId, projectId));
```

Comment:
Please fix the %d , " part.
Also IntelliJ can automatically convert things like "aaa"+b+"ccc" into String.format().

Reply:
Done